### PR TITLE
Incorporate uglify's new object property mangling capabilities

### PR DESCRIPTION
**Pull request by jrhite**
Updated grunt wrapper to be able to hook into uglify's new object property name mangling functionality, the ability to pass files containing property and variable name exceptions and the ability to use a cache to coordinate symbol mangling across multiple calls to uglify.

See uglifyjs v2.4.19 https://www.npmjs.com/package/uglify-js for more info.

### CODE REVIEW

**Comment by vladikoff** [comment]
Thanks for the PR! some tests are failing though :(

**Comment by jrhite** [comment]
The 4 failing tests have been failing since commit: 8f9bb92. Earlier comments from @XhmikosR say that he is on it and fixing them.

I added 5 new tests specific to the new functionality and they're all passing.

**Comment by XhmikosR** [comment]
I'm still waiting for a reply from @vladikoff 'cause as you can see from here https://github.com/gruntjs/grunt-contrib-uglify/compare/tests, I only managed to fix 2...

**Comment by pdachtera** [comment]
+1 for the PR!

**Comment by vladikoff** [comment]
@jrhite could you please rebase your PR? also no need to update the changelog

**Comment by jrhite** [comment]
Rebased PR and reverted changelog changes...

Thanks!

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "grunt-contrib-uglify",
   "description": "Minify files with UglifyJS.",
-  "version": "0.8.1",
+  "version": "0.9.0",
```
Comment (XhmikosR) [inline package.json]: This should be reverted too.

**Comment by chesserik** [comment]
+1 !

**Comment by jk-** [comment]
really good idea! +1

**Comment by yneves** [comment]
best idea ever +1

**Comment by jrhite** [comment]
@XhmikosR I just reverted the package.json version change and rebased to the latest code. Should be good to go.

Cheers!

File: docs/uglify-options.md [hunk 2]
```diff
@@ -139,3 +139,31 @@ Type: `Boolean`
 Default: false
 
 Pass this flag if you don't care about full compliance with Internet Explorer 6-8 quirks.
+
+## mangleProperties
+Type: `Boolean`
```
Comment (XhmikosR) [inline docs/uglify-options.md]: Needs 2 tailing spaces for a new line for all the newly added types.

**Comment by jrhite** [comment]
@XhmikosR good eye again! (uggh...MD requires 2 extra spaces like that...still learning it :-) )

I went ahead and add the 2 trailing spaces to all the new options in docs/uglify-options.md. I also fixed some of the older options that were missing this too.

Checked in and rebased. Thanks!

**Comment by XhmikosR** [comment]
Thanks!

**Comment by jrhite** [comment]
Thanks!

@XhmikosR when/how will this make it to the npm repo? it will be version 0.8.2, correct?

**Comment by XhmikosR** [comment]
I don't make the releases myself so when @vladikoff thinks it's OK :P

Though, perhaps we could wait a couple of days in case we merge some of the other PRs too.

**Comment by jrhite** [comment]
Ah, got it ok. Thanks...

No big rush here, but the sooner the better of course! :-)

**Pull request merged**